src/tests/ftest/pool/Attribute.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+
+import os
+import traceback
+import sys
+import json
+import threading
+import string
+import random
+from avocado       import Test
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+import ServerUtils
+import WriteHostFile
+
+from daos_api import DaosContext, DaosPool, DaosApiError
+
+GLOB_SIGNAL = None
+GLOB_RC = -99000000
+
+def cb_func(event):
+    "Callback Function for asynchronous mode"
+    global GLOB_SIGNAL
+    global GLOB_RC
+    GLOB_RC = event.event.ev_error
+    GLOB_SIGNAL.set()
+
+def verify_list_attr(indata, size, buffer, mode):
+    """
+    verify the length of the Attribute names
+    """
+    # length of all the attribute names
+    aggregate_len = sum(len(attr_name) for attr_name in indata.keys()) + 1
+    # there is a space between each name, so account for that
+    aggregate_len += len(indata.keys())-1
+
+    if aggregate_len != size:
+        raise DaosApiError("FAIL: Size is not matching for Names in list"
```

Comment:
Is this intended to be DaosApiError or DaosTestError? Leaving one note here, usage of the two is mixed throughout. Looks fine besides this.

Reply:
Yes, should not be API error.